packages/tasks/src/local-apps.ts

```diff
@@ -39,36 +45,52 @@ export type LocalApp = {
 			 * And if not (mostly llama.cpp), snippet to copy/paste in your terminal
 			 * Support the placeholder {{GGUF_FILE}} that will be replaced by the gguf file path or the list of available files.
 			 */
-			snippet: (model: ModelData) => string | string[];
+			snippet: (model: ModelData) => Snippet | Snippet[];
 	  }
 );
 
 function isGgufModel(model: ModelData) {
 	return model.tags.includes("gguf");
 }
 
-const snippetLlamacpp = (model: ModelData): string[] => {
+const snippetLlamacpp = (model: ModelData): Snippet[] => {
+	const command = (binary: string) =>
+		[
+			"# Load and run the model:",
+			`${binary} \\`,
+			'  --hf-repo "${model.id}" \\',
+			"  --hf-file {{GGUF_FILE}} \\",
+			'  -p "You are a helpful assistant" \\',
+			"  --conversation",
+		].join("\n");
 	return [
-		`# Option 1: use llama.cpp with brew
-brew install llama.cpp
-
-# Load and run the model
-llama \\
-	--hf-repo "${model.id}" \\
-	--hf-file {{GGUF_FILE}} \\
-	-p "I believe the meaning of life is" \\
-	-n 128`,
-		`# Option 2: build llama.cpp from source with curl support
-git clone https://github.com/ggerganov/llama.cpp.git 
-cd llama.cpp
-LLAMA_CURL=1 make
-
-# Load and run the model
-./main \\
-	--hf-repo "${model.id}" \\
-	-m {{GGUF_FILE}} \\
-	-p "I believe the meaning of life is" \\
-	-n 128`,
+		{
+			title: "Install from brew",
+			setup: "brew install llama.cpp",
+			command: command("llama-cli"),
+		},
+		{
+			title: "Use pre-built binary",
+			setup: [
+				// prettier-ignore
+				"# Download pre-built binary from:",
```

Comment:
In fact, I switched my mind to "UX/UI designer" when I drafted this proposal ;-)

The current UI has a problem that these multiple snippets but no title for them (visually hard to distinct between the 2 snippets):

![image](https://github.com/huggingface/huggingface.js/assets/7702203/e459c1d3-7bf0-4ce7-9a53-11aad47c83b0)

My first iteration would be to have a title for each snippet, then only "expand" one section at a time (while other options are "collapsed")

![image](https://github.com/huggingface/huggingface.js/assets/7702203/aca007f3-f4e4-4d9b-ac96-d6572e7e534b)

But then I think we can also split between the "setup" and "run" step, since ideally the user will setup just once but run multiple times.

Feel free to give other suggestions

Reply:
Good idea I quite like the second image. Let's ask @gary149/ @julien-c for thoughts here

Reply:
cc @gary149 wdyt

Reply:
bumping this as I think it'll be good to merge this soon! cc: @gary149 (sorry for creating an extra notification)

Reply:
okay in principle, but let's see if it's not busted on the Hub side. (sorry for the late reply)